In [43]:
import numpy as np 
def dsin(x:float): return np.sin(np.deg2rad(x))
def dcos(x:float): return np.cos(np.deg2rad(x))
# remember numpy does everything in radians!
g = 9.8 # Gravity
# Find the tension in each of the three cables supporting the traffic light if it weighs 170 N.
# T1 41deg, T2 63deg T_y1sin(41) + T_y2sin(63) = 170N, since F_netx=0 T_x1cos(41)=T_x2cos(63) 
# Find the relation between the two angles: T_x1cos(41)/cos(63) = 1.66239069
# T_y1 sin(41) + (1.662T_y1)sin(63)=170
T_y1 = 170 / (np.sin(np.deg2rad(41))+1.662*np.sin(np.deg2rad(63))) 
T_y2 = T_y1 * 1.66239069
print(f'T_y1 = {T_y1} T_y2 = {T_y2}') 

# A service elevator takes a load of garbage, mass 10.0 kg, from a floor of a skyscraper under 
# construction, down to ground level, accelerating downward at a rate of 1.2 m/s2. Find the 
# magnitude of the force the garbage exerts on the floor of the service elevator.
# F = mg - ma = m(g - a) = (10.0kg)(9.8 - 1.2)
F = 12 * (9.8-1.9)
print(f'F={F}')

# Atwood Machine 
# (m2 - m1)g = (m1 + m2)a
# a=((m2-m1)/(m1+m2))g
m1 = 1.8
m2 = 4.5 
a = g*((m2-m1)/(m1+m2))
print(f'a={a}')
T = m1*(g+a)
print(T)

'''
Two blocks are connected by a massless rope as shown below. The mass of the block on the 
table is 3.2 kg and the hanging mass is 0.8 kg. The table is frictionless, and the pulley 
is frictionless and massless. (a) Find the acceleration of the blocks. (b) Find the tension 
in the rope. (c) Find the speed with which the hanging mass hits the floor if it starts from 
rest and is initially located 1.2 m from the floor.

a = g*((m2)/(m1+m2))
T = g*((m1*m2)/(m1+m2)) 

v^2 = v0^2 + 2a(x - x0) for this:  
v = sqrt(0+2a(1.2 - 0))
'''
m1 = 3.2
m2 = 0.8
a= g*((m2)/(m1+m2))
T = g*((m1*m2)/(m1+m2)) 
v=np.sqrt(2*a*1.2)
print('Question 4:')
print(f'a={a}')
print(f'T={T}')
print(f'v={v}')

'''
A 1.8 kg block (mass 1) and a 3.2 kg block (mass 2) are connected by a light string as shown; 
the inclination of the ramp is 40°. Friction is negligible. What is (a) the acceleration of 
each block and (b) the tension in the string?

This is one where we want to decompose the components of g for m1
m2*g -T = m2 *a
T - m1 * g * sin(theta)=m1 * a
a = ((m2*g-m1*g*sin(theta))/(m1+m2)) 
T = (m1*g*sin(theta)) + (m1 * a) 
'''
print('Question 5:')
m1 = 1.8
m2 = 3.2
a = ((m2*g*dsin(40))/(m1+m2))
T = (m1*g*dsin(40)) + (m1 * a) 
print(f'a={a}')
print(f'T={T}')

'''
At a post office, a parcel that is a 20.0-kg box slides down a ramp inclined at 24.1° 
with the horizontal. The coefficient of kinetic friction between the box and plane is 0.0300.

Find the acceleration of the box: 
R = m1*g*cos(theta)
a = ((m * g * sin(theta)) - (mu * R))/m
Find the velocity of the box as it reaches the end of the plane, if the length of the 
plane is 1.6 m and box starts at rest.

'''
print('Question 6:')
m1 = 20
mu = 0.03 
theta = 24.1
R = m1 * g * dcos(theta)
d=1.6
a = ((m1 * g * dsin(theta)) - (mu * R))/m1
v=np.sqrt(2*a*d)
print(f'a = {a} v = {v}')

'''
Calculate the centripetal force on the end of a 115-m (radius) wind turbine blade that is 
rotating at 0.5 rev/s. Assume the mass is 6 kg.

velocity is: revs * 2 pi /s 

a_c = v^2/r 
F_c = ma_c

rev/s = (pi * r )s 
'''
print('Question 7:')
rev = 0.5
m1 = 6
r = 115
a = np.power((2 * np.pi * rev),2)*r
F= m1 * a 
print(f'F_c = {F}')
# LaTex:
# a_T = \frac{d|\overrightarrow{v}|}{dt}


T_y1 = 79.55405284245595 T_y2 = 132.24991679706682
F=94.80000000000001
a=4.2
25.2
Question 4:
a=1.9600000000000002
T=6.272000000000002
v=2.1688706738761536
Question 5:
a=4.031563887953975
T=18.595588433187707
Question 6:
a = 3.7332652636310586 v = 3.456363528857951
Question 7:
F_c = 6810.027036751657


$ F_k = \mu \cdot F_N $
$ F_N = F_{wy} = F_w \cdot cos(\theta)$  

Chapter 7: Work Formulas

$ dW = \vec{F} \cdot d\vec{x} = |\vec{F}| | d\vec{x} | cos\theta $

A toy cart is pulled a distance of 6.0 m in a straight line across the floor. The force pulling the cart has a magnitude of 28.5 N and is directed at 45° above the horizontal. What is the work done by this force?

$ W = Fd cos\theta = (28.5N)(6m)cos(45^\circ) = 120.9155J$ 

A sled plus passenger with total mass 65 kg is pulled 23 m across the snow ( μk=0.2 ) at a constant velocity by a force directed 25° above the horizontal.

Calculate the work of the applied force: calculate the magnitude of the applied force, then the x

$ W = F_T cos\theta d \text{Need to find } F_T = \frac{mg\mu_k}{cos\theta+\mu_k sin\theta} $

calculate the work of friction: $W = F cos\theta d$

calculate magnitude of friction: $ F = mg\left(\frac{1}{1+\mu_k tan\theta}\right)\mu_k$


calculate total work 

How much work does the force $F(x)=\left( - \frac{2.0}{x} \right) N$ do on a particle as it moves from $x=2.0m$ to $x=5.0m$?

$ W=\int_{x_i}^{x_f} F(x)dx = -2.0 ln\frac{8.0}{2.0} = -2.772588722$

How much work is done against the gravitational force on a 3.5-kg briefcase when it is carried from the ground floor to the roof of the Empire State Building, a vertical climb of 380 m?

$W = 3.5 kg \cdot 9.8 \frac{m}{s^2} \cdot 380m = 13034J$

A bungee cord exerts a nonlinear elastic force of magnitude $F(x)=k_1 x + k_2 x^3 $, where x is the distance the cord is stretched,  
$k_1 = 190 \frac{N}{m}$  and  
$k_2 = -0.223 \frac{N}{m^3}$. How much work must be done on the cord to stretch it 16.7 m?

$ W = \int_{0}^{x_f} F(x)dx = \frac{1}{2}k_1 x^2_f+\frac{1}{4}k_2 x^4_f$

$W=\frac{1}{2}\left( 190\frac{N}{m}\right)\left(16.7m\right)^2+\frac{1}{4}\left( -0.223 \frac{N}{m}\right)\left( 16.7m\right)^4= 22.158.33551kJ$


A 4-kg body has three times the kinetic energy of an 5-kg body. Calculate the ratio of the speeds of these bodies.

An 8.0-g bullet has a speed of 680 m/s.
$K = \frac{1}{2}mv^2 $

$K = \frac{1}{2}0.008\cdot \left(680\frac{m}{s^2} \right) = 1849.6$

Calculate the force needed to bring a 978-kg car to rest from a speed of 99 km/h in a distance of 120 m (a fairly typical distance for a non-panic stop).
99km/h = 27.5 m/s looking for $F=ma$

$v_f^2 = v_0^2 + 2ax \text{ or } a = \frac{v_f^2 - v_i^2}{2x} $

$ a = \frac{(2.75m^s)^2}{2 \cdot 120m} \cdot 978kg = 3081.71875N $

A cyclist in a race must climb a 5° hill at a speed of 7 m/s. If the mass of the bike and the biker together is 61 kg, what must be the power output of the biker to achieve the goal?

$ P = mgv\text{sin}5^\circ  P = 61 \cdot 9.8 \cdot 7 \cdot sin(5) = 364.7119211W$
